pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -151,7 +151,7 @@ func DetectEndpointMode(kubeClient kubelib.Client) EndpointMode {
 	useEndpointslice, ok := features.EnableEndpointSliceController()
 
 	// we have a client, and flag wasn't set explicitly, auto-detect
-	if kubeClient != nil && !ok && kubelib.IsAtLeastVersion(kubeClient, 21) {
+	if kubeClient != nil && !ok && !kubelib.IsLessThanVersion(kubeClient, 21) {
```

Comment:
Why this change? Also can we reuse the same function/check at both places?

Reply:
This is because   when err happens , it may detect the k8s versions error, in this case if the k8s cluster is 1.20, then we could wrongly use endpointslice.
```
func IsAtLeastVersion(client Client, minorVersion uint) bool {
	clusterVersion, err := client.GetKubernetesVersion()
	if err != nil {
		return true
	}
	return IsKubeAtLeastOrLessThanVersion(clusterVersion, minorVersion, true)
}
```